src/grpc_server.rs

```diff
@@ -0,0 +1,253 @@
+            }
+            Err(e) => {
+                format!("Failed to stop application {}: {}", app.project_id, e)
+            }
+        };
+
+        let response = ResponseMessage { message };
+        Ok(Response::new(response))
+    }
+
+    /// # Deployment Status
+    /// Get the status of a deployment from its uuid.
+    /// # Arguments
+    /// The request containing the uuid of the deployment to get the status of.
+    /// # Returns
+    /// Nothing, wrapped in a Result.
+    async fn deployment_status(
+        &self,
+        request: Request<DeploymentStatusRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let deployment_uuid = request.into_inner().deployment_uuid;
+
+        let app = get_app(&self.db, deployment_uuid);
+
+        let status = self
+            .docker_engine
+            .get_application_status(app.project_id.as_str(), app.application_id.as_str())
+            .await
+            .unwrap();
+
+        let message = format!("Application {} is {:?}", app.project_id, status);
+
+        let response = ResponseMessage { message };
+        Ok(Response::new(response))
+    }
+
+    /// # Apply Config Deployment
+    /// Apply a configuration to a deployment from its uuid.
+    /// # Arguments
+    /// The request containing the uuid of the deployment to apply the configuration to, as well as its configuration in JSON format.
+    /// # Returns
+    /// Nothing, wrapped in a Result.
+    async fn apply_config_deployment(
+        &self,
+        request: Request<ApplyConfigDeploymentRequest>,
+    ) -> Result<Response<ResponseMessage>, Status> {
+        let request = request.into_inner();
+        let config = request.config;
+        let deployment_uuid = request.deployment_uuid;
+
+        let deployment_join_project = &self.db.get_deployment_join_project(deployment_uuid.clone());
+
+        let app = Application {
+            application_id: deployment_uuid.clone(),
+            project_id: deployment_join_project.project_uuid.clone(),
+            image_name: format!("{}/{}", deployment_join_project.user_id, deployment_uuid),
+            image_tag: String::from("latest"),
+            env_variables: serde_json::from_str(&config).unwrap(),
+        };
+
+        let message: String = match self.docker_engine.restart_application(&app).await {
+            Ok(_) => match self.db.set_deployment_config(deployment_uuid, config).await {
+                Ok(_) => {
+                    format!("Applied config to application {}", app.project_id)
+                }
+                Err(e) => {
+                    format!(
+                        "Failed to save config to database {}: {}",
+                        app.project_id, e
+                    )
+                }
+            },
+            Err(e) => {
+                format!(
+                    "Failed to apply config to application {}: {}",
+                    app.project_id, e
+                )
+            }
+        };
```

Comment:
It's strange to return every time an ok message and not using the `Err` with `Status`?

Something like this (example for failing to save config in db):

```rust
let msg = format!(
    "Failed to save config to database {}: {}",
    app.project_id, e
);
return Err(Status::new(Code::Internal, msg));
```

I have seen this here and in `start_deployment()`, `restart_deployment()`, `delete_deployment()`, `stop_deployment()`

Reply:
I didn't understood what you want; if the first match is ok then it does db save which in turn can return ok or err (?)

Reply:
it's the goal of the `Result`, to return `Err` if there is an error, but here you're returning an error inside `Ok`